substrate/frame/state-trie-migration/src/lib.rs

```diff
@@ -618,10 +673,16 @@ pub mod pallet {
 
 			// ensure that the migration witness data was correct.
 			if real_size_upper < task.dyn_size {
+				T::Currency::hold(&HoldReason::SlashForContinueMigrate.into(), &who, deposit)?;
```

Comment:
was a bit surprised that there is no fallible direct function to slash as it was in the past, that we have to artifically hold, then slash.